modules/generator/processor/spanmetrics/spanmetrics.go

```diff
@@ -31,11 +33,15 @@ type Processor struct {
 	spanMetricsDurationSeconds registry.Histogram
 	spanMetricsSizeTotal       registry.Counter
 
+	filter *spanfilter.SpanFilter
+
+	discardCounter prometheus.Counter
+
```

Comment:
nit: name it something to make it clear it belongs with span filters? and remove the space between to visually pair them?

`filteredSpansCounter` ? dunno :)